Add getServiceURL selector.

ankitrox opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #8848 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

nfmohit commented on assets/js/modules/reader-revenue-manager/datastore/service.js:
```diff
@@ -0,0 +1,73 @@
+/**
+ * `modules/reader-revenue-manager` data store.
```
```suggestion
 * `modules/reader-revenue-manager` data store: service.
```

nfmohit commented on assets/js/modules/reader-revenue-manager/datastore/service.js:
```diff
@@ -0,0 +1,73 @@
+	 * @param {string} publicationID Publication ID.
+	 * @return {string} Link to Reader Revenue Management platform.
+	 */
+	getServiceURL: createRegistrySelector(
```
A couple points about the selector:

1. In the DocBlock, it says that it accepts a `publicationID` parameter, but in reality, it doesn't appear to do so. This should be accepted as an optional parameter and added to the URL if defined.
2. The arguments object (containing `path` and `query`) are not included in the DocBlock. Please see `assets/js/modules/analytics-4/datastore/service.js` for example.

ankitrox replied:
The docblock has been updated. I think as you added this comment on line 

```js
getServiceURL: createRegistrySelector(
```

which is still unchanged thus GH is showing the this has not been addressed. 

Can you please look at the docblock and let me know if it is fine?

Thanks

nfmohit commented on assets/js/modules/reader-revenue-manager/datastore/service.test.js:
```diff
@@ -0,0 +1,103 @@
+import { CORE_USER } from '../../../googlesitekit/datastore/user/constants';
+import { decodeServiceURL } from '../../../../../tests/js/mock-accountChooserURL-utils';
+
+describe( 'module/reader-revenue-manager service store', () => {
```
```suggestion
describe( 'modules/reader-revenue-manager service store', () => {
```

nfmohit commented on assets/js/modules/reader-revenue-manager/datastore/service.test.js:
```diff
@@ -0,0 +1,103 @@
+				);
+			} );
+
+			it( 'should append `reader-revenue-manager` path to the URL', () => {
```
This sentence feels a little subjective where it may seem expected to have `reader-revenue-manager` as an exact string appended to the URL. I'd suggest a generic test case name:

```suggestion
			it( 'should add the path parameter', () => {
```

nfmohit commented on assets/js/modules/reader-revenue-manager/datastore/service.test.js:
While the publication ID can simply be added to the URL using the `query` sub-parameter, as this is a query parameter that will be used quite often, we should accept that as an optional `publicationID` parameter and add test coverage for it as well.

ankitrox replied:
Added an additional optional `publicationID` param along with `path` and `query` parameters.

nfmohit reviewed (changes requested):
Nice work here, @ankitrox ! Please look at my comments below:

1. Please take a look at my code review comments below.
2. Please update the QAB. It is not necessary to activate the module. At this point, that might not even be successful in many cases as the module is under development. Simply activating the `rrmModule` feature flag should be sufficient.

Please let me know if you have any questions, thank you!

nfmohit reviewed (changes requested):
Hi @ankitrox 👋 All my code review comments were not addressed, so I'm sending this back to you for another look. Please let me know if you have any questions. Thanks.

ankitrox commented:
> Hi @ankitrox 👋 All my code review comments were not addressed, so I'm sending this back to you for another look. Please let me know if you have any questions. Thanks.

Hi @nfmohit 👋🏼 
Sorry for that!
I've addressed couple of comments that went unaddressed. The other one was addressed, but not showing that it has been changed, so I've added [comment](https://github.com/google/site-kit-wp/pull/8953#discussion_r1671678692) in that one.

Over to you for review.

Thanks

nfmohit reviewed (approved):
LGTM, thank you @ankitrox !
